### PR TITLE
add tox for running tests on multiple python versions 

### PR DESCRIPTION
**Pull request by retornam**
@carljm this PR adds a tox.ini file, which takes care of https://github.com/Instagram/MonkeyType/issues/89

### CODE REVIEW

**Comment by retornam** [comment]
@carljm I wasn't able to reproduce your error. I've updated the `tox.ini` file's command section to use `{envbindir}/pytest` . Try that  and lets see if it fixes your issue.

```
pipenv shell
Spawning environment shell (/bin/zsh). Use 'exit' to leave.
 . /Users/retornam/.virtualenvs/MonkeyType-ZbTACiMr/bin/activate
➜  MonkeyType git:(retornam/addtox)  . /Users/retornam/.virtualenvs/MonkeyType-ZbTACiMr/bin/activate
(MonkeyType-ZbTACiMr) ➜  MonkeyType git:(retornam/addtox) tox
GLOB sdist-make: /Users/retornam/src/MonkeyType/setup.py
py36 inst-nodeps: /Users/retornam/src/MonkeyType/.tox/dist/MonkeyType-18.5.2.dev1.zip
py36 installed: atomicwrites==1.1.5,attrs==18.1.0,click==6.7,coverage==4.5.1,Django==2.0.5,MonkeyType==18.5.2.dev1,more-itertools==4.2.0,pluggy==0.6.0,py==1.5.4,pytest==3.6.3,pytest-cov==2.5.1,pytz==2018.5,retype==17.12.0,six==1.11.0,typed-ast==1.1.0
py36 runtests: PYTHONHASHSEED='4160919102'
py36 runtests: commands[0] | /Users/retornam/src/MonkeyType/.tox/py36/bin/pytest
========================================================================= test session starts ==========================================================================
platform darwin -- Python 3.6.5, pytest-3.6.3, py-1.5.4, pluggy-0.6.0
rootdir: /Users/retornam/src/MonkeyType, inifile: pytest.ini
plugins: cov-2.5.1
collected 210 items

demo/test_inbox.py ........                                                                                                                                      [  3%]
tests/test_cli.py ..............                                                                                                                                 [ 10%]
tests/test_config.py .....                                                                                                                                       [ 12%]
tests/test_encoding.py ........................                                                                                                                  [ 24%]
tests/test_stubs.py ...............................................................................                                                              [ 61%]
tests/test_tracing.py ..................s                                                                                                                        [ 70%]
tests/test_typing.py ................................................                                                                                            [ 93%]
tests/test_util.py ........                                                                                                                                      [ 97%]
tests/db/test_base.py .                                                                                                                                          [ 98%]
tests/db/test_sqlite.py ....                                                                                                                                     [100%]

================================================================ 209 passed, 1 skipped in 0.76 seconds =================================================================
_______________________________________________________________________________ summary ________________________________________________________________________________
  py36: commands succeeded
  congratulations :)
```

**Comment by retornam** [comment]
@carljm is this PR still valid?

**Comment by carljm** [comment]
@retornam definitely! Sorry it’s taken me a few days to get back to it. I’ll take a look tomorrow and see if I can figure out what’s going wrong on my machine.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  py36
+
+[testenv]
+deps = 
+  pytest
+  pytest-cov
+  django==2.0.5      
+
+commands =
+  {envbindir}/pytest
```
Comment (carljm) [inline tox.ini]: The `{envbindir}/` here is not necessary, it works for me locally without it. I think my previous problems were due to having an installation of `tox-pipenv` still sitting around locally; once I ran `pipenv clean` things worked.

So let's remove `{envbindir}/`.

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[tox]
+envlist = 
+  py36
+
+[testenv]
+deps =
```
Comment (carljm) [inline tox.ini]: Let's make the only tox dep `pipenv==2018.7.1` and instead add `pipenv sync -d` as a first command in `commands`; this way the dependency information is consolidated in the Pipfile and we don't repeat it here, instead we use pipenv to install the dev dependencies in each toxenv.

**Comment by retornam** [comment]
@carljm your review comments have been addressed in https://github.com/Instagram/MonkeyType/pull/93/commits/f8ba058b7c398cba6c18e1b96fca3db45c93973f

**Comment by carljm** [comment]
Merged! Thanks @retornam for the contribution.

**Pull request merged**